backend/tests/alert/test_alert.py

```diff
@@ -806,9 +806,7 @@ def test_alert_called_wrong_sem(self, mock_alert):
             "sent" in json.loads(res.content)["message"],
         )
         self.assertFalse(mock_alert.called)
-        self.assertEqual(1, StatusUpdate.objects.count())
-        u = StatusUpdate.objects.get()
-        self.assertFalse(u.alert_sent)
```

Comment:
why does this test change?

Reply:
My understanding is this test was kind of flawed from the beginning – it attempts to create a StatusUpdate object with a poorly formatted semester. In the `record_update` function there is a check after the creation of the object that raises an error if the semester is poorly formatted. When changing it to `transaction.atomic()`, throwing this error leads to reverting the creation of StatusUpdate, which makes more sense logically anyways, as a StatusUpdate object with a poorly formatted semester doesn't make sense.